docs/CreateMessage/create-message.plantuml

```diff
@@ -0,0 +1,91 @@
+@startuml create-message
+!pragma teoz true
+
+title Create Message
+
+actor sender as "Sender"
+control fn as "CreateMessage"
+database db as "messages"
+collections st as "processing-message"
+participant ai as "ai-common"
+queue q as "message-created"
+
+sender -> fn ++: //Submit Message//
+
+fn -> fn: Parse the //Request Payload//
```

Comment:
Before this parsing, we should have a series of middleware that check if the sender has the right permissions to send a message with this payload (for example, if they can send advanced messages).

Reply:
@Garma00 where are these middlewares defined? I found only one check on ACLs (https://github.com/pagopa/io-functions-services/blob/master/CreateMessage/handler.ts#L471) after the `create messageId` step

Reply:
[Here](https://github.com/pagopa/io-functions-services/blob/master/CreateMessage/handler.ts#L606) they are

Reply:
thanks!!

do you know why these acls are checked two times?

Reply:
Those are 2 different checs: the `canWriteMessage` ensures that the sender is allowed to send a message to this specific citizen, while the middlewares I linked, check that the sender can send a message with that specific payload